### PR TITLE
Update ASP.NET Core sample to use minimal host

### PR DESCRIPTION
**Pull request by captainsafia**
This updates the ASP.NET Core sample to use some of the "modern" functionality available in .NET 6+ including top-level statements, the new minimal-style `WebApplicationBuilder`, and removal of the `Startup` class.

I've also included an `.http` file to support testing of the endpoint.

As a next step, I'd like to take a stab at replacing the controller class + input formatter with a minimal endpoint + `BindAsync` implementation, but I think this change is helpful for now.

### CODE REVIEW

File: samples/CloudNative.CloudEvents.AspNetCoreSample/CloudNative.CloudEvents.AspNetCoreSample.http [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@HostAddress = https://localhost:5001
+
+POST {{HostAddress}}/api/events/receive
+Content-Type: application/json
+CE-SpecVersion: 1.0
+CE-Type: "com.example.myevent"
+CE-Source: "urn:example-com:mysource:abc"
+CE-Id: "c457b7c5-c038-4be9-98b9-938cb64a4fbf"
```
Comment (captainsafia) [inline samples/CloudNative.CloudEvents.AspNetCoreSample/CloudNative.CloudEvents.AspNetCoreSample.http:8]: In an ideal world, I wouldn't hardcode the GUID here and would instead use dynamic variable support provided by `.http` files. However, it appears that each editor recognizes a different dynamic variable for generating a GUID.

VS Code and VS respect `{{$guid}}` and Rider respects `{{$random.uuid}}` so there is no universal value to include here. Opted to hard code for now.

**Comment by captainsafia** [comment]
> Does anything use the new .http file? (It's not something I've come across before, and I'm not sure whether it's used implicitly or whether it's just as an example that has some IDE support for "send this request".) It might be worth a README file in this directory.

Rider and VS provide built-in support for executing the requests in the `.http` file. VS Code has an [extension](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) that you can use. I'll pop a README in the sample directory that outlines how to run the sample and execute commands against it with the CLI tool and HTTP files.

File: samples/CloudNative.CloudEvents.AspNetCoreSample/Program.cs [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
-using Microsoft.AspNetCore.Hosting;
+using CloudNative.CloudEvents.AspNetCoreSample;
+using Microsoft.AspNetCore.Builder;
+using CloudNative.CloudEvents.NewtonsoftJson;
+using Microsoft.Extensions.DependencyInjection;
 
-namespace CloudNative.CloudEvents.AspNetCoreSample
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers(opts =>
```
[the remark this answers is not in the document]
Comment (captainsafia) [reply samples/CloudNative.CloudEvents.AspNetCoreSample/Program.cs:12]: > Use an expression-bodied lambda here?

Sure! I don't mind using something terser here.

File: samples/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Samples
```
Comment (jskeet) [inline samples/README.md:1]: This is amazing!

**Review by jskeet — APPROVED** [review]
Thanks so much, the readme in particular is great!

**Comment by jskeet** [comment]
Doh - one thing, we need each commit to have a "Signed-off-by" line. 

If you could squash the two commits down to one that contains the Signed-off-by, I'll then merge. Thanks!

**Comment by captainsafia** [comment]
> If you could squash the two commits down to one that contains the Signed-off-by, I'll then merge. Thanks!

Woop -- I ended up signing the second commit and force-pushing. If you'd prefer I squash let me know.

**Comment by jskeet** [comment]
> > If you could squash the two commits down to one that contains the Signed-off-by, I'll then merge. Thanks!
> 
> Woop -- I ended up signing the second commit and force-pushing. If you'd prefer I squash let me know.

Fingers crossed it'll be fine - I'll squash it myself as part of merging, which I *think* should work...

**Comment by captainsafia** [comment]
Build failure! Embarrassing 🙊 I'll fix this

**Pull request merged**